csharp/Platform.Numbers/CalcResult.cs

```diff
@@ -0,0 +1,461 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+
+namespace Platform.Numbers
+{
+    public class CalcResult<T>
+    {
+        public CalcResult() : this(default) { }
+
+        public CalcResult(T value)
+        {
+            Value = value;
+            IsValid = true;
+        }
+
+        #region Arithmetic operations
+      
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public CalcResult<T> Add(T a)
+        {
+            if (IsValid)
```

Comment:
Each `if` statement will add additional performance overhead. And conditional jump instruction is much slower than addition instruction.